src/PostgREST/Request/ApiRequest.hs

```diff
@@ -448,10 +450,46 @@ requestContentTypes conf action target =
     invokeContentTypes =
       defaultContentTypes
         ++ rawContentTypes conf
-        ++ [CTOpenAPI | tpIsRootSpec target]
+        ++ [CTOpenAPI | targetProcIsRootSpec]
     defaultContentTypes =
       [CTApplicationJSON, CTSingularJSON, CTTextCSV]
 
 rawContentTypes :: AppConfig -> [ContentType]
 rawContentTypes AppConfig{..} =
   (ContentType.decodeContentType <$> configRawMediaTypes) `union` [CTOctetStream, CTTextPlain]
+
+{-|
+  Search a pg procedure by its parameters. Since a function can be overloaded, the name is not enough to find it.
+  An overloaded function can have a different volatility or even a different return type.
+-}
+findTargetProc :: QualifiedIdentifier -> S.Set Text -> Bool -> Bool -> ProcsMap -> Either ApiRequestError Target
+findTargetProc qi payloadKeys paramsAsSingleObject isRootSpec allProcs =
+  case bestMatch of
+    []     -> Left $ NoRpc (qiSchema qi) (qiName qi) (S.toList payloadKeys) paramsAsSingleObject
+    [proc] -> Right $ TargetProc proc isRootSpec
+    procs  -> Left $ AmbiguousRpc (toList procs)
```

Comment:
I like how the `Lefts` are now inside the function, that makes the logic easier to understand.

However, it's weird how it returns a `Target` type - which can be a `TargetDefaultSpec`, among other things.

Is it possible to return a `ProcDescription` instead?